docs/howto/object-storage/s3/presign.md

```diff
@@ -0,0 +1,167 @@
+---
+description: You can use the S3 API to temporarily allow access to individual objects without authentication from a browser, using pre-signed URLs.
+---
+# Pre-signed object URLs
+
+{{page.meta.description}}
+
+This approach is distinct from enabling *permanent* [anonymous access to objects in a bucket](public-bucket.md).
+If you are planning to use pre-signed URLs in a bucket, then that bucket should *not* have a public read policy.
+
+> Pre-signed URLs are typically found in web applications using a Software Development Kit (SDK) to interact with the S3 API.
+> An example of such an SDK would be [Boto3](https://boto3.amazonaws.com/v1/documentation/api/latest/index.html) for Python applications.
+>
+> This how-to guide instead uses command-line utilities to illustrate the concept, and to facilitate testing pre-signed URLs in {{brand}}.
+
+## Prerequisites
+
+The use of pre-signed URLs requires that you configure your environment with [working S3-compatible credentials](credentials.md).
+
+## Creating a pre-signed URL
+
+To create a pre-signed URL for an object, you use the following command (replace `<seconds>` with the number of seconds you want the pre-signed URL to be valid):
+
+=== "aws"
+    ```bash
+    aws --profile <region> \
+      --endpoint-url=https://s3-<region>.{{brand_domain}}:8080 \
+      s3 presign \
+      --expires-in <seconds>
+      s3://<bucket-name>/<object-name>
+    ```
+=== "mc"
+    ```bash
+    mc share download --expire <seconds>s <region>/<bucket-name>
+    ```
+    Note the `s` suffix for the `--expire` option.
+    `mc` also supports the suffix `m` for minutes, `h` for hours, and `d` for days.
+
+    If you do not set `--expire`, `mc` defaults to creating a pre-signed URL that is valid for 7 days.
+=== "s3cmd"
+    > For pre-signed URLs generated with `s3cmd` to work correctly in {{brand}}, you must use an `s3cmd` version later than 2.0.1, and set `signurl_use_https = True` in your configuration file.
+
+    ```bash
+    s3cmd -c ~/.s3cfg-<region> signurl s3://<bucket-name>/<object-name> +<seconds>
+    ```
+    Note the `+` prefix on the expiry.
+    `s3cmd` also supports setting an absolute expiry date, which does not use the `+` prefix and must be formatted in [Unix time](https://en.wikipedia.org/wiki/Unix_time).
+
+The command will return the valid pre-signed URL.
+
+
+## Accessing objects with a pre-signed URL
+
+To access an object in a public bucket from a web browser or a generic HTTP/HTTPS client like `curl`, open the URL that the pre-sign command returned:
+
+```console
+curl -f -O https://s3-<region>.{{brand_domain}}:8080/<bucket-name>/<object-name>?AWSAccessKeyId=<access-key>&Signature=<signature>&Expires=<expiry>
+```
+
+As long as the query parameters are correct and the signature has not yet expired, this command will succeed.
+If the query parameters are incorrect or the pre-signed URL is past its expiry date, it will fail with HTTP 403 (Forbidden) instead.
```

Comment:
Instead of "past its expiry date" I would write "past its expiration date".

Reply:
I actually polled native speakers in my circle about this very issue a few months ago, and the preference for "expiry" vs. "expiration" is entirely regional. Since we are not following a particular national standard of English here, I prefer to be concise and use the shorter word.

Reply:
Fair enough!